nautobot/core/graphql/utils.py

```diff
@@ -1,6 +1,68 @@
+import graphene
+
+
+from django_filters.filters import BooleanFilter, NumberFilter, MultipleChoiceFilter
+from nautobot.utilities.filters import MultiValueNumberFilter
+
+
 def str_to_var_name(verbose_name):
     """Convert a string to a variable compatible name.
     Examples:
         IP Addresses > ip_addresses
     """
     return verbose_name.lower().replace(" ", "_").replace("-", "_")
+
+
+def get_filtering_args_from_filterset(filterset):
+    """Generate a list of filter arguments from a filterset.
+
+    The FilterSet class will be instantiated before extracting the list of arguments to
+    account for dynamic filters, inserted when the class is instanciated. (required for Custom Fields filters).
+
+    Filter field that are inheriting from BooleanFilter and NumberFilter will be converted
+    to their appropriate type, everything else will be of type String.
+    if the field field is a subclass of MultipleChoiceFilter, the argument will be converted as a list
+
+    Args:
+        filterset(FilterSet): FilterSet class used to extract the argument
+
+    Returns:
+        dict(graphene.Argument): Filter Arguments organized in a dictionary
+    """
+
+    args = {}
+    instance = filterset()
+
+    for filter_name, filter_field in instance.filters.items():
+
+        field_type = graphene.String
+        filter_field_class = type(filter_field)
+
+        if issubclass(filter_field_class, BooleanFilter):
+            field_type = graphene.Boolean
+
+        elif issubclass(filter_field_class, NumberFilter):
+            field_type = graphene.Int
+
+        if issubclass(filter_field_class, MultiValueNumberFilter):
+            field_type = graphene.List(graphene.Int)
+
+        elif issubclass(filter_field_class, MultipleChoiceFilter):
+            field_type = graphene.List(field_type)
```

Comment:
Would reorganizing like this be any better/clearer?

```python
if issubclass(filter_field_class, MultiValueNumberFilter):
    field_type = graphene.List(graphene.Int)
else:
    if issubclass(filter_field_class, BooleanFilter):
        field_type = graphene.Boolean
    elif issubclass(filter_field_class, NumberFilter):
        field_type = graphene.Int
    else:
        field_type = graphene.String

    if issubclass(filter_field_class, MultipleChoiceFilter):
        field_type = graphene.List(field_type)
```

Reply:
Make sense